src/main/java/com/ericsson/ei/mongo/MongoDBHandler.java

```diff
@@ -363,42 +361,62 @@ private boolean doDrop(String dataBaseName, String collectionName, MongoQuery qu
 
     }
 
-    private MongoCollection<Document> getMongoCollection(String dataBaseName,
+    private MongoCollection<Document> getMongoCollection(String databaseName,
             String collectionName) {
-        if (mongoClient == null)
+        if (mongoClient == null) {
             return null;
-        MongoDatabase db;
+        }
+
+        try {
+            List<String> collectionList = getCollectionList(databaseName);
+            if (!collectionList.contains(collectionName)) {
+                createCollection(databaseName, collectionName);
+            }
+        } catch (MongoClientException e) {
+            LOGGER.error("Failure when handling Mongo collection: {} ", e.getMessage(), e);
+            return null;
+        }
+
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
+        MongoCollection<Document> collection = db.getCollection(collectionName);
+        return collection;
+    }
+
+    private List<String> getCollectionList(String databaseName) {
+        MongoDatabase db = mongoClient.getDatabase(databaseName);
         List<String> collectionList;
 
         try {
-            db = mongoClient.getDatabase(dataBaseName);
             collectionList = db.listCollectionNames().into(new ArrayList<String>());
         } catch (MongoInterruptedException | MongoSocketReadException | MongoSocketWriteException
                 | MongoCommandException | IllegalStateException e) {
-            LOGGER.error("Failed to get Mongo collection, Reason : {} ", e.getMessage(), e);
+            String message = "Failed to get Mongo collection list, Reason: " + e.getMessage();
```

Comment:
You have used String.format before but not here, perhaps use it here too?

Reply:
Done.